src/RA_Achievement.cpp

```diff
@@ -574,225 +606,225 @@ void Achievement::Set(const Achievement& rRHS)
+//  {
+//      nIter++;
+//      fStepAt = fStep * nIter;
+//  }
+//
+//  if( fStepAt >= 1.0f )
+//      fStepAt = 1.0f;
+//
+//  return fStepAt;
 //}
 //
-//float Achievement::ParseProgressExpression( char* pExp )
+// float Achievement::ParseProgressExpression( char* pExp )
 //{
-//	if( pExp == nullptr )
-//		return 0.0f;
-// 
-//	int nOperator = 0;	//	0=add, 1=mult, 2=sub
-// 
-//	while( *pExp == ' ' )
-//		++pExp; //	Trim
-// 
-//	float fProgressValue = 0.0f;
-//	char* pStrIter = &pExp[0];
-// 
-//	int nIterations = 0;
-// 
-//	while( *pStrIter != nullptr && nIterations < 20 )
-//	{
-//		float fNextVal = 0.0f;
-// 
-//		//	Parse operator
-//		if( pStrIter == pExp )
-//		{
-//			//	Start of string: assume operator of 'add'
-//			nOperator = 0;
-//		}
-//		else
-//		{
-//			if( *pStrIter == '+' )
-//				nOperator = 0;
-//			else if( *pStrIter == '*' )
-//				nOperator = 1;
-//			else if( *pStrIter == '-' )
-//				nOperator = 2;
-//			else
-//			{
-//				char buffer[256];
-//				sprintf_s( buffer, 256, "Unrecognised operator character at %d",
-//					(&pStrIter) - (&pExp) );
-// 
-//				ASSERT(!"Unrecognised operator in format expression!");
-//				return 0.0f;
-//			}
-//			
-//			pStrIter++;
-//		}
-// 
-//		//	Parse value:
-//		if( strncmp( pStrIter, "Cond:", 5 ) == 0 )
-//		{
-//			//	Get the specified condition, and the value from it.
-//			unsigned int nCondIter = strtol( pStrIter+5, nullptr, 10 );
-//			
-//			if( nCondIter < NumConditions() )
-//			{
-//				Condition& Cond = GetCondition( nCondIter );
-//				if( Cond.CompSource().m_nVarType != CMPTYPE_VALUE )
-//				{
-//					fNextVal = (float)Cond.CompSource().m_nVal;
-//				}
-//				else if( Cond.CompTarget().m_nVarType != CMPTYPE_VALUE )
-//				{
-//					//	Best guess?
-//					fNextVal = (float)Cond.CompTarget().m_nVal;
-//				}
-//				else
-//				{
-//					//wtf? Both elements are 'value'?
-//					ASSERT(!"Bolloxed");
-//					fNextVal = 0.0f;
-//				}
-//			}
-//			else
-//			{
-//				//	Report error?
-//				ASSERT(!"Bolloxed2");
-//				return 0.0f;
-//			}
-//		}
-//		else if( pStrIter[0] == '0' && pStrIter[1] == 'x' )
-//		{
-//			CompVariable Var;							//	Memory location
-//			Var.ParseVariable( pStrIter );				//	Automatically pushes pStrIter
-//			fNextVal = (float)( Var.GetValue() );
-//		}
-//		else
-//		{
-//			//	Assume value, assume base 10
-//			fNextVal = (float)( strtol( pStrIter, &pStrIter, 10 ) );
-//		}
-// 
-//		switch( nOperator )
-//		{
-//		case 0:	//	Add
-//			fProgressValue += fNextVal;
-//			break;
-//		case 1:	//	Mult
-//			fProgressValue *= fNextVal;
-//			break;
-//		case 2:	//	Sub
-//			fProgressValue -= fNextVal;
-//			break;
-//		default:
-//			ASSERT(!"Unrecognised operator?!");
-//			break;
-//		}
-// 
-//		while( *pExp == ' ' )
-//			++pExp; //	Trim any whitespace
-// 
-//		nIterations++;
-//	}
-// 
-//	if( nIterations == 20 )
-//	{
-//		ASSERT(!"Bugger... can't parse this 'progress' thing. Too many iterations!");
-//	}
-// 
-//	return fProgressValue;
```

Comment:
Should we just get rid of all this commented out code? It'll reduce the lines "added" significantly.